### PR TITLE
#8 create a contributingmd which explains how to contribute to this repository

### LINKED ISSUE #8: Create a Contributing.md which explains how to contribute to this repository.

### PR DESCRIPTION
**Pull request by Nutstree**
closes #8

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+Please note if you are working on a certain issue then make sure to stay active with development.
+
+## Git Commit, Branch, and PR Naming Conventions
+
+When you are working with git, please be sure to follow the conventions below on your pull requests, branches, and commits:
+
+```text
+PR: [#ISSUE ID] Title of the PR
+Branch: [ISSUE ID]-title-of-the-pr (shorter)
+Commit: [[ISSUE ID]] what was done
```
Comment (Zegveld) [inline CONTRIBUTING.md:14]: Commits should be #[IssueId] so that it becomes a link. For example: `#8` becomes #8

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+If you want to work on a new feature, follow these steps.
+
+1. Create or choose an issue
+2. Fork the repo
+3. Clone your fork
+4. Checkout a new branch
+5. Do your work
+6. Commit
+7. Push your branch to your fork
+8. Go into github UI and create a PR from your fork & branch, and merge it into upstream MAIN
```
Comment (Zegveld) [inline CONTRIBUTING.md:49]: Merge is done by maintainers, not by contributers.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+When contributing to this repository, please first discuss the change you wish to make via [issues](https://github.com/rigd-loxia/builder-generator/issues).
+
+Please note if you are working on a certain issue then make sure to stay active with development.
+
+## Git Commit, Branch, and PR Naming Conventions
+
+When you are working with git, please be sure to follow the conventions below on your pull requests, branches, and commits:
+
+```text
+PR: [#ISSUE ID] Title of the PR
```
Comment (Zegveld) [inline CONTRIBUTING.md:12]: PR text should contain a line saying:
closing #[IssueId]
To link the issue that is closed by this PR correctly.

**Review by Zegveld — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (Nutstree) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 2 above)
Comment (Nutstree) [reply CONTRIBUTING.md]: removed the merge instructions

File: CONTRIBUTING.md (hunk 3 above)
Comment (Nutstree) [reply CONTRIBUTING.md]: changed PR to PR Title and PR description to make it more clear

**Review by Zegveld — APPROVED** [review]

**Pull request merged**